hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/subscription/id/NotificationAwareSubscriptionIdsCache.java

```diff
@@ -0,0 +1,58 @@
+package pl.allegro.tech.hermes.consumers.subscription.id;
+
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.consumers.subscription.cache.SubscriptionsCache;
+import pl.allegro.tech.hermes.domain.notifications.InternalNotificationsBus;
+import pl.allegro.tech.hermes.domain.notifications.SubscriptionCallback;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class NotificationAwareSubscriptionIdsCache implements SubscriptionIds, SubscriptionCallback {
+
+    private final Map<SubscriptionId, SubscriptionName> ids = new HashMap<>();
```

Comment:
Don't we have to care for thread safety?

Reply:
Hmm it supposed to be used by a single thread, but as long as it's not used anywhere yet, I think it should allow concurrent access. Good point, I'll fix it, thanks. :)

Reply:
Fixed.